### PR TITLE
leap year

### PR DESCRIPTION
**Pull request by Subhransupanda2000**

### CODE REVIEW

File: src/main/java/com/example/jspiderjava/MapExample.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package com.example.jspiderjava;
+
+import java.util.concurrent.ConcurrentHashMap;
+
+public class MapExample {
+    public static void main(String[] args) {
+        ConcurrentHashMap ch=new ConcurrentHashMap<>();
```
Comment (rkpatra201) [inline src/main/java/com/example/jspiderjava/MapExample.java:7]: Why we used concurrenthashmap instead hashmap

File: src/main/java/com/example/jspiderjava/ReverseString.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package com.example.jspiderjava;
+
+public class ReverseString {
+    public static void main(String[] args) {
+        String string="hello";
+        StringBuffer sb=new StringBuffer(string);
+        sb.reverse();
```
Comment (rkpatra201) [inline src/main/java/com/example/jspiderjava/ReverseString.java:7]: How to reverse a string without using loop or above method

**Review by rkpatra201 — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/jspiderjava/MapExample.java (hunk 1 above)
Comment (Subhransupanda2000) [reply src/main/java/com/example/jspiderjava/MapExample.java:7]: Hashmap is not thread safe . If we will make it threadsafe its perfomance will go down. concurrenthasmap solving all this using  16 boquite approach

File: src/main/java/com/example/jspiderjava/ReverseString.java (hunk 2 above)
Comment (Subhransupanda2000) [reply src/main/java/com/example/jspiderjava/ReverseString.java:7]: I have added that in this pr

File: src/main/java/com/example/jspiderjava/MapExample.java (hunk 1 above)
Comment (rkpatra201) [reply src/main/java/com/example/jspiderjava/MapExample.java:7]: But here your code is running in single thread and always thread safe

**Review by rkpatra201 — CHANGES_REQUESTED** [review]
Ok

File: src/main/java/com/example/jspiderjava/ReverseStringMethod2.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        char[] arr = st.toCharArray();
+        String result = reverseString(arr);
+        System.out.println(result);
+
+    }
+
+    public static String reverseString(char[] arr) {
+        int left = 0;
+        int right = arr.length - 1;
+        while (left < right) {
```
Comment (rkpatra201) [inline src/main/java/com/example/jspiderjava/ReverseStringMethod2.java:15]: This is loop using while

**Review by rkpatra201 — CHANGES_REQUESTED** [review]

**Pull request merged**